Checkout: Update Thank You page copy for non-domain purchases

stephanethomas opened the pull request:
This pull request addresses part of #2484 by updating the feature sections of the `Thank You` page with the new copy for chargeback, site redirect, Google Apps, and Jetpack plans purchases:

![screenshot](https://cloud.githubusercontent.com/assets/594356/13393088/892c4938-dedf-11e5-8fa7-658064a0067c.png)

#### Testing instructions
 
1. Run `git checkout update/thank-you-copy` and start your server
2. Purchase a [site redirect](http://calypso.localhost:3000/domains/add/site-redirect/YOURSITE) and check the `Thank You` page
3. Purchase [Google Apps](http://calypso.localhost:3000/domains/add/site-redirect/YOURSITE) for an existing domain registration and check the `Thank You` page
4. Purchase a chargeback and check the `Thank You` page
5. Switch to a Jetpack site
6. Purchase a [Jetpack plan](http://calypso.localhost:3000/plans/YOURSITE) and check the `Thank You` page
 
#### Additional notes

Jetpack plans cannot be tested until https://github.com/Automattic/wp-calypso/pull/3617 is deployed. Icons are addressed in https://github.com/Automattic/wp-calypso/pull/3578.

#### Reviews
 
- [x] Code
- [x] Product

scruffian commented on client/my-sites/upgrades/checkout-thank-you/business-plan-details.jsx:
```diff
@@ -10,33 +10,31 @@ import i18n from 'lib/mixins/i18n';
+				title={ i18n.translate( 'Start a Live Chat' ) }
+				description={ i18n.translate( 'Have a question? Chat live with WordPress.com Happiness Engineers.' ) }
+				buttonText={ i18n.translate( 'Talk to an Operator' ) }
+				href="//support.wordpress.com/live-chat/"
```
Are there any cases where this shouldn't be https? Will `//` work from inside the desktop app?

stephanethomas replied:
> Are there any cases where this shouldn't be https?

I don't think so but we have already quite a few urls that are protocole relative in our code anyway. My point is that we should be consistent, so whatever format we choose we stick to it. This is something that can be addressed in a separate pull request though.

> Will `//` work from inside the desktop app?

@mjangda confirmed that me should be fine here.

stephanethomas replied:
Upon some thoughts, we're going to standardize all these urls in a separate pull requests.

scruffian commented on client/my-sites/upgrades/checkout-thank-you/purchase-detail.jsx:
```diff
@@ -9,47 +9,50 @@ import React from 'react';
-		'is-placeholder': isPlaceholder
-	} );
+const PurchaseDetail = React.createClass( {
+	propTypes: {
```
Is it necessary to change this to a stateful component?

stephanethomas replied:
Well that was my understanding since we have now a `renderButton()` function.

scruffian replied:
Do you think its worth making it stateful just to have this extra function? Couldn't we encapsulate that in an `if` statement?

drewblaisdell replied:
I agree with @scruffian - we have other options here that don't involve converting this to a stateful component:

- `renderButton` could be a pure function at the module scope, instead of a method.
- We could assign the button JSX to a `buttonElement` variable if `buttonText` is present.

The latter is probably better. We _do_ want to use the stateful component syntax when we would otherwise be creating a new function on every render (like when we pass a prop to an event handler with bind()), but that is not the case here.

scruffian commented on client/my-sites/upgrades/checkout-thank-you/business-plan-details.jsx:
```diff
@@ -10,33 +10,31 @@ import i18n from 'lib/mixins/i18n';
-					}
-					buttonText={ i18n.translate( 'Claim your free domain' ) }
-					href={ '/domains/add/' + selectedSite.slug } />
+				additionalClass="live-chat"
```
Do we still want to promote live chat in this case? It's not in the doc, so we should probably ask @mikeshelton1503.

stephanethomas replied:
That's something I raised internally in p6qnuF-1Y5-p2#comment-2562.

fabianapsimoes commented:
#### Site Redirect

- The Site Redirect sub-header doesn't match the copy:

It says:
> You will receive an email confirmation shortly for your purchase of Site Redirect. What's next?

Should be:
> Your site is now redirecting to example.com. It’s doing somersaults in excitement!

- Also where it says "Get started with your site's new features" it should probably be "What now?

- Missing ", from the Domains screen." in the second CTA. 

- The "Manage redirect" button seems broken. It doesn't leave the Thank You screen, even though it changes the URL in the address bar. The path seems wrong. 

In the button it is:
`http://calypso.localhost:3000/domains/manage/fabianasimoes.com/edit/fabianasignup20151028151955dot.com`

In the address bar it shows:
`http://calypso.localhost:3000/domains/manage/fabianasimoes.com/edit`

Should be:
`http://calypso.localhost:3000/domains/manage/fabianasimoes.com/redirect-settings/fabianasignup20151028151955dotcom.wordpress.com`

fabianapsimoes commented:
#### Google Apps

Sub-header doesn't match the copy. It is: 
> You will receive an email confirmation shortly for your purchase of Extra licenses for Google Apps. What's next?

Should be:
> Your domain example.com is now set up to use Google Apps. It’s doing somersaults in excitement!

Right now it says "Get started with your site's new features" below the banner. We don't have copy for that, but I assume it should be "What now?" since the box describe steps the user must follow.

"Finish setting up your Google Apps account" is missing the required treatment. Are we doing this in a separate PR?

fabianapsimoes commented:
#### Jetpack Plans

Hm, I just bought Business for a Jetpack site but the Thank You page shows the copy for WordPress.com Business (instead of the copy for Jetpack Plans):

![screenshot-calypso localhost 3000 2016-02-29 14-11-49](https://cloud.githubusercontent.com/assets/1248436/13395347/86c1bd68-deee-11e5-8fd9-d528091aad17.png)

Site was blog.tugdev.net, and had Jetpack Premium before, if that makes any difference.

fabianapsimoes commented:
I'm not quite sure how to test chargebacks, so I pinged @peterbutler for help. He's at a meetup this week, so I wouldn't block on it. We can fix any issues separately.

stephanethomas commented:
Thanks for the thorough review @fabianapsimoes!

Headers (and sub-headers) as well as catch phrases (i.e. `What now?`) will be updated in a different pull request. This one only takes care of the main content of the page.

#### Site redirect

> Missing ", from the Domains screen." in the second CTA. 

I removed this part because it sounded awkward to me. I don't think we ever referenced our pages like this anywhere in Calypso. Moreover it is incorrect since we're talking of the `Domain Settings` page here. I should have brought that point up internally though - I completely forgot about it.

> The "Manage redirect" button seems broken. It doesn't leave the Thank You screen, even though it changes the URL in the address bar. 

I added a missing protocol to the button. Could you give it another try?

> The path seems wrong.

I don't think so. It makes sense - at least to me - to redirect users on the `Domain Settings` page since we also offer them to update the primary domain:

> Disable the redirect by choosing a different primary domain, or change the target address, from the Domains screen.

#### Google Apps

> "Finish setting up your Google Apps account" is missing the required treatment. 

Good catch! I had a quick look at adding the yellowish banner but since it wasn't trivial I'll handle that in a different pull request.

#### Jetpack plans

I think that's because https://github.com/Automattic/wp-calypso/pull/3617 wasn't deployed. It is now and I rebased this pull request so you can give it another spin.

fabianapsimoes commented:
> I removed this part because it sounded awkward to me. I don't think we ever referenced our pages like this anywhere in Calypso.

Sounds good to me :ok_hand: 

> I don't think so. It makes sense - at least to me - to redirect users on the Domain Settings page since we also offer them to update the primary domain:

Yeah I thought it was maybe the path that caused the broken button, but it was the protocol :) That location sounds good to me.

Product :+1:

drewblaisdell commented on client/my-sites/upgrades/checkout-thank-you/site-redirect-details.jsx:
```diff
@@ -10,29 +10,38 @@ import { getDomainManagementUrl } from './utils';
+					)
+				}
+				buttonText={ i18n.translate( 'Try it now' ) }
+				href={ `http://${ domain }` }
```
Is there a reason that you switched to `http://${ domain }` instead of whatever the selected site's `URL` property is? Are we sure that `selectedSite.URL` always uses this protocol?

stephanethomas replied:
> Is there a reason that you switched to http://${ domain } instead of whatever the selected site's URL property is?

I had to use `domain` because `selectedSite.URL` would still point to the previous primary domain until the list of sites is refreshed.

> Are we sure that selectedSite.URL always uses this protocol?

We don't support `https` for site redirect so `http` is fine.

stephanethomas replied:
I had another look to our internal document with the new copy and it seems this button should point to the default address of the site. I'll update that.

drewblaisdell commented:
I updated the commit history to not turn `PurchaseDetail` into a stateful component, in favor of conditionally assigning a button to a variable inside its render(). I also fixed a bug where the placeholder state was off when `buttonText` was missing. Feel free to force push over my changes if they were made erroneously, or if there are better solutions.

If, however, you're cool with that change and there is a good reason for [this change](https://github.com/Automattic/wp-calypso/pull/3633#discussion-diff-54499721), this LGTM.

stephanethomas commented:
Thanks for your help Drew! I had another look to this pull request today and:

* Rebased against `master`
* Removed sections that should no longer appear
* Fixed a few gridicons
* Updated the url for the redirect button

I then tested again purchases of:

- [x] Business plan
- [x] Chargeback
- [x] Generic
- [x] Google Apps
- [x] Jetpack plan
- [x] Premium plan  
- [x] Premium plan free trial
- [x] Site redirect

While testing free trials though I realized that users never see the `Thank You` page since they are redirected on the `Plan Overview` page instead. Therefore I decided to remove checks that prevent the free domain section from being displayed in case of free trial in the `Thank You` page, since that's just dead code.

scruffian commented on client/my-sites/upgrades/checkout-thank-you/site-redirect-details.jsx:
```diff
@@ -10,29 +10,38 @@ import { getDomainManagementUrl } from './utils';
+					)
+				}
+				buttonText={ i18n.translate( 'Try it now' ) }
+				href={ `http://${ selectedSite.wpcom_url }` }
```
Good change :+1:

scruffian commented on client/my-sites/upgrades/checkout-thank-you/site-redirect-details.jsx:
```diff
@@ -10,29 +10,38 @@ import { getDomainManagementUrl } from './utils';
-				description={ i18n.translate( 'You can disable the redirect or change the target at any time.' ) }
-				buttonText={ i18n.translate( 'My Domains' ) }
-				href={ getDomainManagementUrl( selectedSite ) } />
+				description={ i18n.translate( 'Disable the redirect by choosing a different primary domain, or change the target address.' ) }
```
In the doc this is "Disable the redirect by choosing a different primary domain, or change the target address, from the Domains screen."

stephanethomas replied:
Thanks for catching that. That's something we decided [not to add though](https://github.com/Automattic/wp-calypso/pull/3633/#issuecomment-190254745).

scruffian commented:
LGTM :+1:
